src/GraphQl/Type/FieldsBuilder.php

```diff
@@ -522,7 +522,10 @@ private function convertType(Type $type, bool $input, Operation $rootOperation,
         }
 
         if ($this->typeBuilder->isCollection($type)) {
-            return $this->pagination->isGraphQlEnabled($rootOperation) && !$input ? $this->typeBuilder->getResourcePaginatedCollectionType($graphqlType, $resourceClass, $rootOperation) : GraphQLType::listOf($graphqlType);
+            $resourceMetadataCollection = $this->resourceMetadataCollectionFactory->create($resourceClass);
+            $resolverOperation = $resourceMetadataCollection->getOperation(null, true, false, true);
```

Comment:
can't you use an operation name instead of adding a flag?

Reply:
? Not sure to understand your point sorry.